server/sonar-server/src/main/java/org/sonar/server/platform/db/EmbeddedDatabaseFactory.java

```diff
@@ -31,19 +31,19 @@
 
   private static final String URL_PREFIX = "jdbc:h2:tcp:";
 
-  private final Settings settings;
+  private final Configuration config;
   private final System2 system2;
   private EmbeddedDatabase embeddedDatabase;
 
-  public EmbeddedDatabaseFactory(Settings settings, System2 system2) {
-    this.settings = settings;
+  public EmbeddedDatabaseFactory(Configuration config, System2 system2) {
+    this.config = config;
     this.system2 = system2;
   }
 
   @Override
   public void start() {
     if (embeddedDatabase == null) {
-      String jdbcUrl = settings.getString(DatabaseProperties.PROP_URL);
+      String jdbcUrl = config.get(DatabaseProperties.PROP_URL).get();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655)